### PR TITLE
Changed value for window flow control.

### PR DESCRIPTION
**Pull request by theSage21**
Changed default value for flow control everywhere.

### CODE REVIEW

**Comment by theSage21** [comment]
I think I changed too many things. Were all of those window control values?

**Comment by Lukasa** [comment]
@theSage21 You did change too many things, but that's fine, I now have something to give feedback on! I'll make notes inline in the diff. =)

File: hyper/http20/connection.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def __init__(self, host, port=None, secure=None, window_manager=None, enable_pus
         #: network. This is used as a performance optimisation. Increase buffer
         #: size to improve performance: decrease it to conserve memory.
         #: Defaults to 64kB.
-        self.network_buffer_size = 65536
+        self.network_buffer_size = 2**24
```
Comment (Lukasa) [inline hyper/http20/connection.py]: This is not a flow control value, it's just an in-memory buffer to store TCP data. We can leave this as-is for now.

File: hyper/http20/connection.py [hunk 2]
```diff
@@ -122,17 +122,17 @@ def __init_state(self):
...
-            SettingsFrame.INITIAL_WINDOW_SIZE: 65535,
+            SettingsFrame.INITIAL_WINDOW_SIZE: 2**24,
         }
 
         # The socket used to send data.
         self._sock = None
 
         # The inbound and outbound flow control windows.
-        self._out_flow_control_window = 65535
+        self._out_flow_control_window = 2**24
```
Comment (Lukasa) [inline hyper/http20/connection.py]: This *is* a flow control window, but it's not one we control: the remote side of the connection owns it. We can't just change it's value.

File: hyper/http20/stream.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def __init__(self,
         # There are two flow control windows: one for data we're sending,
         # one for data being sent to us.
         self._in_window_manager = window_manager
-        self._out_flow_control_window = 65535
+        self._out_flow_control_window = 2**24
```
Comment (Lukasa) [inline hyper/http20/stream.py]: Again, this *is* a flow control window, but not one we control: the remote side of the connection owns it.

File: hyper/http20/connection.py [hunk 4]
```diff
@@ -122,17 +122,17 @@ def __init_state(self):
 
         # Values for the settings used on an HTTP/2 connection.
         self._settings = {
-            SettingsFrame.INITIAL_WINDOW_SIZE: 65535,
+            SettingsFrame.INITIAL_WINDOW_SIZE: 2**24,
```
Comment (Lukasa) [inline hyper/http20/connection.py]: This setting keeps track of the *remote* side's flow control window, so we can't just change it.

**Comment by Lukasa** [comment]
Ok, the parts of the diff I commented on are the parts we don't need. =)

**Comment by Lukasa** [comment]
Sorry @theSage21, I missed that you updated this!

This looks great, we're missing one thing: we need to update the settings we initially send to the remote peer during negotiation. In `HTTP20Connection._send_preamble`, we need to add a line that sets the INITIAL_WINDOW_SIZE settings to 2**24. Mind adding that?

**Comment by theSage21** [comment]
@Lukasa I am out for some exams right now. Will add as soon as I get back to a laptop. Is this what I should be adding? `f.settings[SettingsFrame.INITIAL_WINDOW_SIZE] = 2**24` Aah, I feel so noobish. I need to read more.

**Comment by Lukasa** [comment]
@theSage21 No rush, take whatever time you need. =) And no worries about feeling that way, it's always hard to come in and work on a new codebase.

Yeah, that's what you'll want to add. Once that's done we'll need to fix up these test failures as well, but let's do one thing at a time. ;)

**Comment by theSage21** [comment]
@Lukasa It seems that I have a lot of ground to cover up. I am closing this PR for now. I will ask for a pull once I am confident enough about what I am doing. Thanks for the help. :smile:

**Comment by Lukasa** [comment]
Ok @theSage21, let me know if you need any assistance!

**Pull request closed without merging**